src/neuroconv/tools/aws/_submit_aws_batch_job.py

```diff
@@ -0,0 +1,640 @@
+        raise ValueError(f"Job queue '{job_queue_status}' failed to launch after {max_retries} retries.")
+
+    return None
+
+
+def _generate_job_definition_name(
+    *,
+    docker_image: str,
+    minimum_worker_ram_in_gib: int,
+    minimum_worker_cpus: int,
+) -> str:  # pragma: no cover
+    """
+    Generate a job definition name for the AWS Batch job.
+
+    Note that Docker images don't strictly require a tag to be pulled or used - 'latest' is always used by default.
+
+    Parameters
+    ----------
+    docker_image : str
+        The name of the Docker image to use for the job.
+    minimum_worker_ram_in_gib : int
+        The minimum amount of base worker memory required to run this job.
+        Determines the EC2 instance type selected by the automatic 'best fit' selector.
+        Recommended to be several GiB to allow comfortable buffer space for data chunk iterators.
+    minimum_worker_cpus : int
+        The minimum number of CPUs required to run this job.
+        A minimum of 4 is required, even if only one will be used in the actual process.
+    """
+    docker_tags = docker_image.split(":")[1:]
+    docker_tag = docker_tags[0] if len(docker_tags) > 1 else None
+    parsed_docker_image_name = docker_image.replace(":", "-")  # AWS Batch does not allow colons in job definition names
+
+    job_definition_name = f"neuroconv_batch"
+    job_definition_name += f"_{parsed_docker_image_name}-image"
+    job_definition_name += f"_{minimum_worker_ram_in_gib}-GiB-RAM"
+    job_definition_name += f"_{minimum_worker_cpus}-CPU"
+    if docker_tag is None or docker_tag == "latest":
+        date = datetime.now().strftime("%Y-%m-%d")
+        job_definition_name += f"_created-on-{date}"
+
+    return job_definition_name
+
+
+def _ensure_job_definition_exists_and_get_arn(
+    *,
+    job_definition_name: str,
+    docker_image: str,
+    minimum_worker_ram_in_gib: int,
+    minimum_worker_cpus: int,
+    role_info: dict,
+    batch_client: "boto3.client.Batch",
+    max_retries: int = 12,
+) -> str:  # pragma: no cover
+    """
+    Ensure that the job definition exists in AWS Batch and return the official ARN.
+
+    Automatically generates a job definition name using the docker image, its tags, and worker configuration.
+    The creation date is also appended if the docker image was not tagged or was tagged as 'latest'.
+
+    Note that registering a job definition does not require either a compute environment or a job queue to exist.
+
+    Also note that registration is strict; once created, a job definition is tagged with ':< revision index >' and
+    any subsequent changes create a new revision index (and technically a new definition as well).
+
+    Parameters
+    ----------
+    job_definition_name : str
+        The name of the job definition to use for the job.
+    docker_image : str
+        The name of the Docker image to use for the job.
+    minimum_worker_ram_in_gib : int
```

Comment:
Nope - there's a comment in the code about the serialization. It really should be an int always. I could relax and make it `MiB`, but I really think the conversion from `GiB` is a lot more convenient since you will always want RAM in that schedule